### PR TITLE
Introduce client specific events

### PR DESCRIPTION
**Pull request by rpkamp**
| Q                | A
| ---------------- | -----
| Bug fix          | no
| New feature      | yes
| BC breaks        | no
| Deprecations     | yes
| Tests pass       | yes
| Fixed tickets    | #283
| License          | MIT

### CODE REVIEW

**Comment by gregurco** [comment]
Hello @rpkamp 

First, thanks for PR :) but, let's do some small changes. You mixed here deprecations and new way of events, but in master we accumulate changes just for v8. These changes will not enter in v7. I propose to create 2 PRs:
- first PR should be created from v7 branch (https://github.com/8p/EightPointsGuzzleBundle/tree/v7). In that PR only deprecation messaged will be added, to notify the user that events system will be changed in v8. After that commit I will create version v7.6.1.
- secund PR is the current one, created on master. Here you can remove all deprecation, old interface and add just new code, as it will be in v8.

Is it ok for you?

Thanks, Vlad.

**Comment by rpkamp** [comment]
I think Symfony normally already adds the new way of doing something, while deprecating the old thing, and then remove the old thing in the new major. The nice thing about this is that you can already update the new thing in the current major and then for the new major you don't have to do anything anymore. 

For this example this would mean that this PR goes against v7, and then in another PR the deprecations will be removed for v8. The advantage is that there is a clean upgrade path from v7 to v8. If we only deprecate in v7, then once you upgrade to v8 your code is temporarily broken until you fix it. If v7 already supports the new way you can get rid of deprecations there and know it will work in v8.

Up to you of course, just offering an alternative :slightly_smiling_face:

**Comment by gregurco** [comment]
@rpkamp I got your idea. If you want this PR to appear in v7, then change PR to be for v7 branch and not for master branch. All what we have on master will be released as v8.

**Comment by rpkamp** [comment]
@gregurco I've completely removed the CompilerPass from this PR, so it ready to be merged into `master`. Next I will create a PR against v7 with the same changes plus a deprecation of the old behaviour.

File: src/Resources/doc/intercept-request-and-response.md [hunk 1]
```diff
@@ -1,105 +1,146 @@
...
-        api_partners:
-            base_url: "http://partners.tld"
+        payment:
+            base_url: 'http://api.payment.example'
+        crm:
+            base_url: 'http://api.crm.tld'
 ```
 
-and `api_partners` requires authorization using some token in header.  
+And suppose that `api_partners` requires authorization using some token in the request header.
```
Comment (gregurco) [inline src/Resources/doc/intercept-request-and-response.md]: Replace here `api_partners` with `payment`

File: src/Resources/doc/intercept-request-and-response.md [hunk 2]
```diff
@@ -1,105 +1,146 @@
...
 How can we do that?
 
-## Interceptor class
+## Event listener for intercepting requests
 
-First of all we have to create out interceptor class, where all the logic will be:
+First of all we have to write an event listener, that will see al requests from our `payment`
```
Comment (gregurco) [inline src/Resources/doc/intercept-request-and-response.md]: typo in `al`

**Comment by rpkamp** [comment]
All done @gregurco

**Pull request merged**